### PR TITLE
update go and telegraf

### PR DESCRIPTION
**Pull request by pfrcks**
This pull request primarily focuses on updating various dependencies and versions in the codebase. The most significant changes include updating the Go version used in GitHub Actions and in the `go.mod` files, updating the ignored vulnerabilities in the `.trivyignore` file, and updating the Telegraf version in both Linux and Windows setup scripts.

Version updates:

* [`.github/workflows/run_unit_tests.yml`](diffhunk://#diff-394fcfd605d72aeb92748c54045208916c0dd638a4c1e09f2b15cc4ceffbcc48L16-R19): Updated the Go version used in GitHub Actions from 1.19.x to 1.22.x.
* `source/plugins/go/input/go.mod` and `source/plugins/go/src/go.mod`: Updated the Go version from 1.19 to 1.22 in these `go.mod` files. [[1]](diffhunk://#diff-dec75be1d43c9293f45a8af5d251c23b4eb8a98dbfb850d68367738d09aa14cbL3-R3) [[2]](diffhunk://#diff-4c8eb2bbf5d4a6f8f0d84217e96f7fb96f5d919c8f8e2519e5f99e86cd3f9262L3-R3)

Vulnerability management:

* [`.trivyignore`](diffhunk://#diff-a2b39041fe7520e8037b4e596d015ea4825043e383f3fa5f85d5ac7f06f4c177L2-L38): Updated the list of ignored vulnerabilities, removing some and adding others.

Telegraf version updates:

* [`kubernetes/linux/setup.sh`](diffhunk://#diff-41e13b2078423187d4de7802d178ff603e09a6aec1c0c653dbe02834cfd15315L62-R62): Updated the Telegraf version from 1.28.5 to 1.29.4 in the Linux setup script.
* [`kubernetes/windows/setup.ps1`](diffhunk://#diff-5ec4ab2461b56d161a16acc7831e62fa863591ac9e6986aaf6833a9af73f881eL45-R45): Updated the Telegraf download link to point to version 1.29.4 instead of 1.24.2 in the Windows setup script.

### CODE REVIEW

File: kubernetes/windows/setup.ps1 [hunk 1]
```diff
@@ -42,7 +42,7 @@ Write-Host ('Finished Installing Fluentbit')
 Write-Host ('Installing Telegraf');
 try {
     # For next telegraf update, make sure to update config changes in telegraf.conf, tomlparser-prom-customconfig.rb and tomlparser-osm-config.rb
-    $telegrafUri='https://dl.influxdata.com/telegraf/releases/telegraf-1.24.2_windows_amd64.zip'
+    $telegrafUri='https://dl.influxdata.com/telegraf/releases/telegraf-1.29.4_windows_amd64.zip'
```
Comment (ganga1980) [inline kubernetes/windows/setup.ps1:45]: is this version working? in the past, we had issues with telegraf update  on windows hence we didnt upgrade the version.

File: source/plugins/go/input/go.mod [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module Docker-Provider/source/plugins/go/input
 
-go 1.19
+go 1.22
```
Comment (ganga1980) [inline source/plugins/go/input/go.mod]: can you please test this change? if its not done already.

File: kubernetes/linux/setup.sh [hunk 3]
```diff
@@ -59,7 +59,7 @@ sudo tdnf install jq-1.6-1.cm2 -y
 #used to setcaps for ruby process to read /proc/env
 sudo tdnf install libcap -y
 
-sudo tdnf install telegraf-1.28.5 -y
+sudo tdnf install telegraf-1.29.4 -y
```
Comment (ganga1980) [inline kubernetes/linux/setup.sh:62]: Please validate the perf and functional tests for this update, and also review any issues on this telegraf version related to promethus, disk, diskio plugins.

Comment (pfrcks) [reply kubernetes/linux/setup.sh:62]: lot of work to individually test all this. I was hoping to use CI/CD clusters for this. WDYT?

Comment (ganga1980) [reply kubernetes/linux/setup.sh:62]: > lot of work to individually test all this. I was hoping to use CI/CD clusters for this. WDYT?

Can you please run some basic tests on both Linux and Windows to ensure this doesnt have any breaking change before we validate on CI/CD clusters. Primarily for windows, we have seen issues  when we upgraded telegraf.

**Pull request closed without merging**